lib/model/autocomplete.dart

```diff
@@ -289,6 +538,38 @@ class MentionAutocompleteView extends ChangeNotifier {
         streamId: streamId, senderId: userB.userId));
   }
 
+  @override
+  Iterable<Object> getSortedItemsToTest(MentionAutocompleteQuery query) {
+    switch (query) {
+      case UserMentionAutocompleteQuery():
+        return sortedUsers;
+      case ChannelMentionAutocompleteQuery(:var raw):
```

Comment:
Instead of putting conditional logic within `MentionAutocompleteView` to handle both @-mentions and channels/topics, let's leave that class focused on @-mentions and make a new class for channels/topics.

I guess the existing "mention" naming made that design ambiguous, because I do say "#-mention" for mentioning a channel or topic. But the intent of MentionAutocompleteView and MentionAutocompleteQuery is that they're about @-mentions.

Probably a good prep commit is to rename all the MentionAutocomplete… classes to say AtMentionAutocomplete… instead, to make that clearer. ("User mention" isn't quite right, because the same autocomplete interaction can produce a mention of a group or of a wildcard like `@channel`.)

Similarly the new classes can say HashMentionAutocomplete… — "channel mention" isn't quite right because these also include mentioning a topic.

Reply:
Thanks @gnprice! I'd like to get you're suggestion in that case as we'd have two view models for the same autocomplete field, which is not implemented in the current design.

So do you think it'd be OK to update `AutocompleteField` so it has a list of `AutocompleteViewModel` and then, based on the query select and use one of them to search for results? Or would you take another approach?

Reply:
The autocomplete view model's lifetime corresponds to the interaction the user has to attempt an autocomplete. There should be only one of those going on at a time — we won't want to try to show @-mention and #-mention results at the same time, for example — so there should only need to be one autocomplete view model for the field at a time.

The `AutocompleteIntent` will specify which kind of autocomplete is desired. So that can be used to construct the right type of `AutocompleteView`. When there isn't (yet) an `AutocompleteIntent`, we don't make an `AutocompleteView`.